module/zfs/zfs_vfsops.c

```diff
@@ -1701,8 +1701,7 @@ zfs_resume_fs(zfs_sb_t *zsb, const char *osname)
 		 * Since we couldn't setup the sa framework, try to force
 		 * unmount this file system.
 		 */
-		if (zsb->z_os)
-			(void) zfs_umount(zsb->z_sb);
+		(void) zfs_umount(zsb->z_sb);
```

Comment:
The fact that zsb->z_os can change outside a lock is definitely racy and problematic.  Rather than needlessly change this let's leave it as is for now and I'll make a note in the Coverity analysis.  @heary-cao go ahead and drop this hunk when you next refresh this patch.